removed sudo. In order to do that I gave myself the neccessary permis…

HNam05 opened the pull request:
…sions with group membership

# Summary

Provide a brief description of the change and the reasoning behind it.

---

# Reviewers

## Required Reviewers
- @username1  
- @username2  

## Optional Reviewers
- @username3  
- @username4  

(Feel free to add/remove as needed.)

---

# What Changed?

Describe the main changes in this PR:
-
-
-

---

# Testing

Describe how you verified functionality:

- [ ] Unit tests added
- [ ] Existing tests pass
- [ ] Manual testing performed

**Steps to reproduce/test:**
1.  
2.  
3.  

---

# Code Quality Checklist

Before requesting review, ensure:
- [ ] I ran **`make format`**
- [ ] I ran **`make test`**
- [ ] All CI checks pass
- [ ] Code follows project style & conventions
- [ ] New functions/classes include docstrings
- [ ] Public APIs are typed (type hints)

---

# Documentation

- [ ] Code comments updated
- [ ] README updated (if needed)

---

# Additional Notes

- Anything else reviewers should know?
- Attach outputs, plots, logs, or GIFs here.

HNam05 committed:
removed sudo. In order to do that I gave myself the neccessary permissions with group membership

HNam05 committed:
Add graceful hardware failure handling and improve test coverage

- Remove sudo requirement from 'make app' command
- Add connection failure handling with warning logs instead of exceptions
- Implement communication health check to detect powered-off/disconnected motors
- Track consecutive response failures and warn when motor stops responding
- Exit gracefully when hardware is unavailable or not communicating
- Add comprehensive unit tests for connection and communication handling
- Increase test coverage from 55% to 67%

Tsmorz commented on Makefile:
```diff
@@ -43,7 +43,7 @@ docker:
 
 app:
-	sudo ./.venv/bin/python -m motor_python
+	./.venv/bin/python -m motor_python
```
Can you update it to this instead incase our .venv changes name?
```
uv run python -m motor_python
```

HNam05 committed:
no_hardware changes

HNam05 committed:
Merge pull request #31 from TUM-Aries-Lab/no_hardware

Add graceful hardware failure handling and improve test coverage

HNam05 committed:
Add FaultCode IntEnum and motor control examples

- Convert FAULT_CODES dict to FaultCode IntEnum class with get_description() method
- Update motor_status_parser to use FaultCode IntEnum
- Add examples.py with motor control functions (position, velocity, duty cycle, current control)
- Add run_motor_loop for continuous testing of all control modes

HNam05 committed:
Integrate examples motor control loop into main

- Import run_motor_loop from examples.py
- Replace hardcoded test with cycling through all 4 control modes
- Continuous loop tests position, velocity, duty cycle, and current control

HNam05 committed:
Add Path support and complete pathlib migration

- Add Path | str type hint to motor port parameter
- Convert Path to str for serial library compatibility
- Replace os.path with pathlib in conftest.py
- All code review tasks from yesterday now complete

HNam05 committed:
Implement all supervisor code review comments

- Refactor constants to frozen dataclasses (MotorDefaults, FrameBytes, CRCConstants, ScaleFactors, PayloadSizes, MotorLimits, ConversionFactors)
- Rename _build_message -> _build_frame and _send_message -> _send_frame for consistency
- Rename variables: velocity_erpm -> velocity_electrical_rpm, speed_erpm -> motor_speed_erpm
- Replace max(min()) with np.clip() for all value clamping
- Add comprehensive docstrings explaining scale factors and unit conversions
- Move control functions to examples.py with parameterized inputs
- Simplify loop logic using itertools.cycle instead of modulo arithmetic
- Rename mode_func -> control_mode_func
- Combine duplicate if statements in __main__.py
- Remove duplicate self.connected assignments
- Change response -> status in get_status() method
- Use pathlib.Path throughout for cross-platform compatibility
- Remove duplicate dependencies from pyproject.toml
- Define test fixtures in conftest.py (nonexistent_port, test_port, mock_response)
- Remove 'is True/False' from assertions
- All tests passing with 69% coverage
